src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -475,44 +474,50 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     }
     const duration = this.props.graphDuration.value;
     // Keep a map of namespace x promises in order not to fetch several times the same data per namespace
-    const healthPerNamespace = new Map<String, Promise<NamespaceHealth>>();
+    const appHealthPerNamespace = new Map<String, Promise<NamespaceAppHealth>>();
+    const wkldHealthPerNamespace = new Map<String, Promise<NamespaceWorkloadHealth>>();
     // Asynchronously fetch health
     cy.nodes().forEach(ele => {
       const namespace = ele.data('namespace');
       const app = ele.data('app');
       const workload = ele.data('workload');
 
-      if ((workload !== 'unknown' || app !== 'unknown') && (ele.data('isGroup') || !ele.data('parent'))) {
-        const appBasedHealth = this.props.graphType === GraphType.APP && app !== 'unknown';
-        let promise = healthPerNamespace.get(namespace);
+      if (app !== undefined && app !== 'unknown' && (ele.data('isGroup') || !ele.data('parent'))) {
```

Comment:
You should rely on the cy's global scratch data to determine the graph type, don't assume graph type based on node details. At minimum it will be consistent with the rest of the graph code.